compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala

```diff
@@ -1209,20 +1209,29 @@ object messages {
            |""".stripMargin
   }
 
-  case class OverloadedOrRecursiveMethodNeedsResultType(tree: Names.TermName)(implicit ctx: Context)
+  case class OverloadedOrRecursiveMethodNeedsResultType private (termName: String)(implicit ctx: Context)
```

Comment:
Why make the name private? If you're an IDE, wouldn't you want to be able to extract the name from the message?